cmd/argoexec/commands/data.go

```diff
@@ -3,18 +3,39 @@ package commands
 import (
 	"context"
 
+	log "github.com/sirupsen/logrus"
 	"github.com/spf13/cobra"
 )
 
 func NewDataCommand() *cobra.Command {
 	command := cobra.Command{
 		Use:   "data",
 		Short: "Process data",
-		RunE: func(cmd *cobra.Command, args []string) error {
-			ctx := context.Background()
-			wfExecutor := initExecutor()
-			return wfExecutor.Data(ctx)
+		Run: func(cmd *cobra.Command, args []string) {
+			ctx := cmd.Context()
+			err := execData(ctx)
+			if err != nil {
+				log.Fatalf("%+v", err)
+			}
 		},
 	}
 	return &command
 }
+
+func execData(ctx context.Context) error {
+	wfExecutor := initExecutor()
+	defer wfExecutor.HandleError(ctx)
```

Comment:
in wait.go, `HandleError()` uses `bgCtx` - do you see a need to do it differently?

Reply:
(I do see that resource.go is using `ctx` but maybe that's just because it was never modified to use `bgctx`?

Reply:
Oh, I guess it looks like `HandleError()` isn't even using the context that's passed in at all. I'm surprised that the linter is okay with that. I kind of wonder if we should remove it from the method parameter, or if not, make it consistently passed in between all in case it's used in the future.

Reply:
Good discovery,
I have unified to bgCtx.